### ISSUE TITLE
Consider removing redundant step from Lab 7 Exercise 4 instructions

### ISSUE THREAD

**Issue by wanecek** [lunduniversity/introprog#379]
In [Lab 7 exercise 4,](https://github.com/lunduniversity/introprog/blob/master/compendium/modules/w07-setmap-lab.tex#L102-L111) where a string representing a text is to be converted to a Vector of words, the first two steps aim to replace any non-letter character with a standard character (`' '`) to enable one-step splitting (`string.split(' ')`) of the string into words.

As far as I understand, the [first step](https://github.com/lunduniversity/introprog/blob/master/compendium/modules/w07-setmap-lab.tex#L105) (replace any characters which pass the predicate `isWhitespace` with a space (`' '`)) is made redundant by the [second step](https://github.com/lunduniversity/introprog/blob/master/compendium/modules/w07-setmap-lab.tex#L106) (replace any characters which fail the predicate `isLetter` to a space (`' '`)).

This is because there are no 16-bit characters that pass the test of being both a whitespace- and letter-character.

```scala
scala> (Char.MinValue to Char.MaxValue).find(c => c.toChar.isLetter && c.toChar.isWhitespace)
res1: Option[Char] = None
```

If the first step does indeed serve a specific purpose, maybe it would be worth making note of it as a footnote? :)

**Comment by bjornregnell** [comment]
The first step makes e.g. `'\n'` become `' '`. The second step makes e.g. `'1` a space. For example:
```scala

scala> '1'.isLetter
res3: Boolean = false

scala> val s1 = "ab\nc1två3".map(c => if (c.isWhitespace) ' ' else c)
s1: String = ab c1två3

scala> val s2 = s1.map(c => if (!c.isLetter) ' ' else c)
s2: String = "ab c två "
```
Or have I misunderstod what you meant?

**Comment by bjornregnell** [comment]
Perhaps you overlooked the word *falskt* in the second step of the assignment?

**Comment by hnrklssn** [comment]
I believe what he means is that if `c.isWhitespace` is true, then `c.isLetter` is always going to be false (as shown in his exhaustive proof), rendering the whitespace check redundant.

**Comment by bjornregnell** [comment]
Aha! Yes of course, thanks @hnrklssn . So with my example:
```scala
scala> val step1and2 = "ab\nc1två3".map(c => if (!c.isLetter) ' ' else c)
step1and2: String = "ab c två "
```
@wanecek Feel free to come with a PR that merges the steps and revise the explanation of what to do and why. Or du you want me toi fix it?

**Comment by wanecek** [comment]
@hnrklssn Thanks for clarifying :smile: 

@bjornregnell Exactly! In simpler terms: as far as I understand, there are no whitespace characters that would not be replaced when replacing any `!_.isLetter` characters. 

I opened a PR at #380. PTAL when you have time :smile:
